hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java

```diff
@@ -413,7 +412,22 @@ public void checkKey() throws Exception {
 
 
     String snapshotKeyPrefix = createSnapshot(volumeName, bucketName);
-    ozoneBucket.deleteKey(key1);
+
+    GenericTestUtils.waitFor(() -> {
+      try {
+        int keyCount = keyCount(ozoneBucket, key1);
+        if (keyCount == 0) {
+          return true;
+        }
+
+        ozoneBucket.deleteKey(key1);
```

Comment:
For some reason, sometimes two exactly the same keys (including names - not sure if that is something that should happen) would be created in a bucket even if intention was for a single key to be created. I tried different approaches to fix this test but only this one seems to be working. An idea is to wrap deletion of the key inside `waitFor` method and deleting the key until `keyCount` method returns that there are no keys left in a bucket.

**This solution might be masking underlaying issue.** I am sure that key creation in a bucket with FSO layout is well tested, but I am not sure how this could have happened. Whoever is reviewing this, pelase consider this.

Reply:
1. 
> For some reason, sometimes two exactly the same keys (including names - not sure if that is something that should happen) would be created in a bucket even if intention was for a single key to be created.

It is surprising because that should not happen ideally. May be add random numbers or char in dir and key name at line number 404 and 405.

2. Just a thought if it is because of soft delete? You can try following to verify it.

```
    ozoneBucket.deleteKey(key1);

    GenericTestUtils.waitFor(() -> {
      try {
        return keyCount(ozoneBucket, key1) == 0;
      } catch (Exception e) {
        return  false;
      }
    }, 1000, 10000);
```

Reply:
Thanks @hemantk-12! I will try this out.

Reply:
I did exacly how you suggested. However, it did not work. Please have a look at [CI](https://github.com/mladjan-gadzic/ozone/actions/runs/5389078718/jobs/9782557417#step:4:5698) and [commit](https://github.com/mladjan-gadzic/ozone/commit/7db461503c6bb17a787bd0f246858d3489b25331).

Reply:
Thanks for trying the suggestion. Then it could be because HDDS-8526](https://issues.apache.org/jira/browse/HDDS-8526), as Attila mentioned.